clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3226,6 +3231,18 @@ private int processKafkaDataMessage(
         // update checksum for this PUT message if needed.
         partitionConsumptionState.maybeUpdateExpectedChecksum(keyBytes, put);
 
+        if (metricsEnabled && recordLevelMetricEnabled.get()
+            && put.getSchemaId() == AvroProtocolDefinition.CHUNKED_VALUE_MANIFEST.getCurrentProtocolVersion()) {
```

Comment:
I noticed `getCurrentProtocolVersion` returns the `schemaId` which is not intuitive. Is this something that is within the scope of this PR to rename it? If not, feel free to resolve the comment.

Reply:
I certainly agree, and that's certainly something that I could do in this PR, but I would feel uncomfortable doing it without someone like @FelixGV okaying it first. Perhaps there is a reason it needs to remain as `currentProtocolVersion`, as it could be more generalized than just our usage as`schemaId`?

Reply:
Let us not block this PR for this discussion. Maybe we can take this offline.